src/LinkDotNet.NCronJob/Configuration/ConcurrencyConfig.cs

```diff
@@ -0,0 +1,8 @@
+namespace LinkDotNet.NCronJob;
+
+/// <summary>
+/// Defines the configuration settings for managing concurrency within the application.
+/// This record is used to configure how many operations can run in parallel.
+/// </summary>
+internal record ConcurrencyConfig(int MaxDegreeOfParallelism);
```

Comment:
For the sake of simplicity we can totally remove that.
I would argue in >80% of cases you either want to have unlimited concurrency or just one single instance of a job.

Reply:
The reason I added this was to use it to evolve concurrency specific settings. I figured you can add things like `TaskQueueLimit` to control memory usage and managing backpressure, PriorityQueueFilter, or some criteria to manage urgency influencing order of execution, TaskTimeout, and maybe even some policies for throttling (e.g. to prevent calls to external APIs from exceeding backpressure) and retry behavior.
I can remove it if you still feel that simpler is better. However regarding unlimited concurrency, I do think we still need some type of ceiling to prevent accidental over-utilization or resource exhaustion.

Reply:
Renamed this to ConcurrencySettings and updated the comments for better clarity. Now, it explicitly indicates that these settings apply to the scheduler's concurrency (all jobs).